src/utils/composer-helper.js

```diff
@@ -0,0 +1,42 @@
+const applyField = (origin, patch) => {
```

Comment:
We should be careful to support multiple types for a certain config property. It may bring unnecessary complexity to projection developers.

Reply:
You may consider make the patch a `Function`. For each use case, you know the type of the property, and you can pass in the suitable function (probably this function is not needed any more). E.g. for your `applyValue`
```js
const applyValue = (defaults, obj) => Object.keys(obj).reduce((memo, key) =>
  Object.assign(memo, {
    [key]: (obj[key] || i => i)(memo[key]),
  }), Object.clone(defaults));
```